### DISCUSSION TITLE
How to Access $get variable from Create Record ?

### DISCUSSION THREAD

**Discussion by hazwan91** [filamentphp/filament#12856] [Help]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

How to get an access to $get variable within the create record? 
I need to set a condition whether to show the `form actions` based on the search result if the `account type` is government, search has been done and the id no. is valid.
If the account type is a government, instead of filling the form, i want to do the search from the existing database and fill the information. Thus before user do the searching or the id no. is not valid, the form actions are suppose to be not visible yet.

![Screenshot 2024-05-19 084344](https://github.com/filamentphp/filament/assets/97268905/a32020f9-863c-49a0-b2d3-fd148ce8f9ca)

![Screenshot 2024-05-19 084107](https://github.com/filamentphp/filament/assets/97268905/0f873b62-fa87-4cac-b977-eb9fc4505f6d)

I try to make the logic inside the create record since the form actions are in this place. However i can't import any sort of function such as Get $get, Livewire Component $livewire and so on..

Need help.

**Comment by leandrocfe** [comment]
`$this->data`
?

**Reply by hazwan91** [reply]
Right...How can i miss that...Thanks...

**Comment by hazwan91** [comment]
Wait.. There seems to be a problem.. When i choose the government, the script `dd` not triggered...but when i choose non government, the dd triggered... it seems to have some delayed...

UserResource, where the radio button
![Screenshot 2024-05-19 102119](https://github.com/filamentphp/filament/assets/97268905/2ec9ce9f-b2d6-4dfc-8cd3-bdb385cdf8b9)

CreateRecord
![Screenshot 2024-05-19 102156](https://github.com/filamentphp/filament/assets/97268905/6063bbd0-b641-46af-a542-74a543854182)

**Reply by hazwan91** [reply]
Extra info....even without the dd() function, the `form actions` hide on non government while visible on government type...Weird because i think i set my condition correctly.. ignoring the valid condition, even this condition `$this->type['type'] === 'sm2` is seems to have triggered incorrectly

**Comment by hazwan91** [comment]
Ok i found out that if i use the method `getCreateFormAction`, `getCreateAnotherFormAction` and `getCancelFormAction` instead of `getFormActions` it works well by using the `visible` method...No delay or whatsoever. However this would also means, i need to write 3 methods for those 3 buttons which makes it a bit of pain.
